middlewares/protectAuth.js

```diff
@@ -0,0 +1,18 @@
+const jwt = require("jsonwebtoken");
+
+const protectAuth = (req, res, next) => {
+  if (!req.headers.authorization) {
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+
+  const token = req.headers.authorization.split(" ")[1];
+  jwt.verify(token, process.env.JWT_SECRET, (err, decodedData) => {
```

Comment:
what if req.headers.authorization.split(" ")[1] - undefined? (just asking)

Reply:
ill add that in the first if check

Reply:
Maybe jwt.verify can handle this case, I dont know. But it's ok now